src/devices/cpu/sh/sh7014_mtu.cpp

```diff
@@ -0,0 +1,570 @@
+{
+	SH7014_MTU_CHANNEL(config, m_chan[0], DERIVED_CLOCK(1, 1), m_intc,
+		0, // channel
+		sh7014_intc_device::INT_VECTOR_MTU_TGI0A,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI0B,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI0C,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI0D,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI0V,
+		-1
+	);
+
+	SH7014_MTU_CHANNEL(config, m_chan[1], DERIVED_CLOCK(1, 1), m_intc,
+		1, // channel
+		sh7014_intc_device::INT_VECTOR_MTU_TGI1A,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI1B,
+		-1,
+		-1,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI1V,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI1U
+	);
+
+	SH7014_MTU_CHANNEL(config, m_chan[2], DERIVED_CLOCK(1, 1), m_intc,
+		2, // channel
+		sh7014_intc_device::INT_VECTOR_MTU_TGI2A,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI2B,
+		-1,
+		-1,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI2V,
+		sh7014_intc_device::INT_VECTOR_MTU_TGI2U
+	);
+}
+
+void sh7014_mtu_device::map(address_map &map)
+{
+	map(0x00, 0x00).rw(FUNC(sh7014_mtu_device::tstr_r), FUNC(sh7014_mtu_device::tstr_w));
+	map(0x01, 0x01).rw(FUNC(sh7014_mtu_device::tsyr_r), FUNC(sh7014_mtu_device::tsyr_w));
+
+	map(0x20, 0x3f).m(m_chan[0], FUNC(sh7014_mtu_channel_device::map_chan0));
+	map(0x40, 0x5f).m(m_chan[1], FUNC(sh7014_mtu_channel_device::map_chan1_2));
+	map(0x60, 0x7f).m(m_chan[2], FUNC(sh7014_mtu_channel_device::map_chan1_2));
+}
+
+///
+
+uint8_t sh7014_mtu_device::tstr_r()
+{
+	return m_chan[0]->is_enabled()
+	 | (m_chan[1]->is_enabled() << 1)
+	 | (m_chan[2]->is_enabled() << 2);
+}
+
+void sh7014_mtu_device::tstr_w(uint8_t data)
+{
+	LOG("%s: tstr_w %02x\n", machine().describe_context().c_str(), data);
+
+	m_chan[0]->set_enable((data & TSTR_CST0) != 0);
+	m_chan[1]->set_enable((data & TSTR_CST1) != 0);
+	m_chan[2]->set_enable((data & TSTR_CST2) != 0);
+}
+
+uint8_t sh7014_mtu_device::tsyr_r()
+{
+	return m_tsyr & (TSYR_SYNC0 | TSYR_SYNC1 | TSYR_SYNC2);
+}
+
+void sh7014_mtu_device::tsyr_w(uint8_t data)
+{
+	LOG("%s: tsyr_w %02x\n", machine().describe_context().c_str(), data);
+
+	m_tsyr = data;
+}
+
+
+//////////////////
+
+sh7014_mtu_channel_device::sh7014_mtu_channel_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, SH7014_MTU_CHANNEL, tag, owner, clock)
+	, m_intc(*this, finder_base::DUMMY_TAG)
+	, m_timer(nullptr)
+{
+}
+
+void sh7014_mtu_channel_device::device_start()
+{
+	save_item(NAME(m_tcr));
+	save_item(NAME(m_tmdr));
+	save_item(NAME(m_tiorh));
+	save_item(NAME(m_tiorl));
+	save_item(NAME(m_tier));
+	save_item(NAME(m_tsr));
+	save_item(NAME(m_tcnt));
+	save_item(NAME(m_tgr));
+
+	save_item(NAME(m_last_clock_update));
+	save_item(NAME(m_clock_type));
+	save_item(NAME(m_clock_divider));
+	save_item(NAME(m_channel_active));
+	save_item(NAME(m_phase));
+	save_item(NAME(m_counter_cycle));
+	save_item(NAME(m_tgr_clearing));
+
+	m_timer = timer_alloc(FUNC(sh7014_mtu_channel_device::timer_callback), this);
+	m_timer->adjust(attotime::never);
```

Comment:
Adjusting the timer to never fire is pretty pointless as you haven’t set it yet.  Did you mean to do that in `device_reset` rather than in `device_start`?

Reply:
Oops, you're right. Not useful to stop it in device_start. Fixed.